dask/base.py

```diff
@@ -648,7 +649,10 @@ def tokenize(*args, **kwargs):
     """
     if kwargs:
         args = args + (kwargs,)
-    return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
+    time_taken, result = get_wall_time(lambda: tuple(map(normalize_token, args)))
+    if time_taken > 2: # seconds
```

Comment:
This threshold should be configurable. I'm not sure what the best key would be though...